cpp/include/raft/core/math.hpp

```diff
@@ -151,16 +243,89 @@ RAFT_INLINE_FUNCTION auto exp(T x)
 #endif
 }
 
+#if _RAFT_HAS_CUDA
+template <typename T>
+RAFT_DEVICE_INLINE_FUNCTION typename std::enable_if<std::is_same<T, __half>::value, __half>::type
+exp(T x)
+{
+#if (__CUDA_ARCH__ >= 530)
+  return ::hexp(x);
+#else
+  // static_assert(false) would be evaluated during host compilation stage while __CUDA_ARCH__ is at
+  // device compilation stage Using this sizeof(T) != sizeof(T) makes it work as it's only triggered
+  // during template instantiation and thus at device compilation stage
```

Comment:
This comment appears in many places, can you please change it for a one-liner? Such as:
```c++
// Fail during template instantiation if the compute capability doesn't support this operation
```

or remove it altogether